bodo/hiframes/table.py

```diff
@@ -223,6 +241,23 @@ def as_array(self):
     def key(self):
         return self.arr_types, self.has_runtime_cols
 
+    def unify(self, typingctx, other):
+        """Unify two TableType instances (required for runtime join filter typing)."""
+        if (
+            isinstance(other, TableType)
+            and (len(self.arr_types) == len(other.arr_types))
+            and (self.has_runtime_cols == other.has_runtime_cols)
```

Comment:
QQ: Does the provided implementation work properly for runtime array columns (e.g. unify DataFrame types after a pivot was performed). If not it might be safe to replace this with `and not  self.has_runtime_cols and not other.has_runtime_cols` and include a TODO to do a deeper dive for this edge case later.

Reply:
Not sure if it would work. Disabled for now as suggested.